scripts/tuple_tree_generator/generators/cpp.py

```diff
@@ -22,9 +22,11 @@ def __init__(
         schema: Schema,
         base_namespace: str,
         emit_tracking: bool,
+        string_types=[],
         user_include_path: Optional[str] = None,
     ):
         self.schema = schema
+        self.string_types = string_types
         self.emit_tracking = emit_tracking
```

Comment:
These changes are hard to review, I don't see any usage of this.
The commit does not specify the reason for this change nor there is docs.

As I mention other times, these changes would be better suited to be in the same commit of their first user.

Anyway, from what I see, this seems to assume there's a C++ method `fromString`.
What if we ditch all of this and just assume there's a trait in C++ you *have to* implement?
We could use this to handle `std::string` (which would return itself), `MetaAddress` and `enums`.

I'd be very unhappy if we had to add yet another argument to the already obscure CMake machinery. Let's strive to prune complexity.